src/Database/TigerBeetle/Raw/Client.hs

```diff
@@ -107,34 +113,27 @@ setupCompletionCallback handle = \ctx packetPtr _timestamp resultPtr resultLen -
         -- TODO: Come up with a better way to log this
         putStrLn "Warning: Received callback for unknown request context"
 
-data WithClientOps =
-  WithClientOps
-    { onInitFailure :: TBInitStatus -> IO ()
-    , onDeinit :: TBClientStatus -> IO ()
-    , useSubmit :: (TBPacket -> IO TBClientStatus) -> IO ()
-    }
-
 withClient
   :: ClientConfig
   -> ClusterId
   -> Text
-  -> (ClientHandle -> IO a)
-  -> IO ()
+  -> (ClientState -> IO a)
```

Comment:
We could come up with a higher level interface that wraps this function and that uses `MonadIO m => ReaderT ClientState  m a`